### ISSUE TITLE
Permissions needed for cross resource group deployment using DTL?

### ISSUE THREAD

**Issue by ChrisLGardner** [Azure/azure-devtestlab#288]
When deploying a template which updates a resource in another resource group I'm seeing the below error (this was working previously following the fix in #260). Is there a specific user/group I need to grant permissions to for this to work?

```
The template deployment failed with error: 'Authorization failed for template resource 'UpdateVNetDNS' of type 'Microsoft.Resources/deployments'. 
The client '<ClientGUID>' with object id '<ClientGUID>' does not have permission to perform action 'Microsoft.Resources/deployments/write' at scope '/subscriptions/<SubscriptionGUID>/resourceGroups/<ResourceGroupName>/providers/Microsoft.Resources/deployments/UpdateVNetDNS'.'.
```

**Comment by jaymarf** [comment]
Hi, 
I will look into this issue today and update my findings.

**Comment by jaymarf** [comment]
Hi, Chris could you send me the template you tried to use  ?

**Comment by ChrisLGardner** [comment]
Sent.

**Comment by jaymarf** [comment]
The problem is that trying to edit resources using the environment deployment is not allowed. When using environment and arm template deployment it must always be to deploy a new resource. The service principal does not have permissions to access any other resource.

**Comment by ChrisLGardner** [comment]
Following the fix in #260 for cross resource group deployment it worked fine for a while. I'm not sure when the change happened as I haven't tested it in a while but there was definitely a period where it worked as I expected it to.

Does the process in DTL differ from deploying it using PowerShell? I'm assuming the PS deployment uses the logged in users credentials rather than the service principle. 

Is there a way to grant the service principle permissions to edit existing resources? Either at the resource group level or the individual resource level?
